.github/workflows/reusable-docker-build.yml

```
@@ -0,0 +1,346 @@
+        if: inputs.dry_run
+        uses: docker/login-action@b45d80f862d83dbcd57f89517bcf500b2ab88fb2 # v4.0.0
+        env:
+          AWS_REGION: us-east-1
+        with:
+          registry: ${{ env[inputs.dry_run_ecr_repo_env] || env.PRIVATE_ECR_DRY_RUN_REPO }}
+          username: ${{ steps.configure-aws-credentials-prod.outputs.aws_access_key_id }}
+          password: ${{ steps.configure-aws-credentials-prod.outputs.aws_secret_access_key }}
+
+      - name: Generate Docker Tags
+        id: generate-tags
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:40:24: Double quote to prevent globbing and word splitting

(shellcheck)